### ISSUE TITLE
How to embed images based on dev-1.0?

### ISSUE THREAD

**Issue by bluesky2091** [memspace/zefyr#478]
The current examples about how to embed images are based on the 0.x version: https://zefyr-editor.gitbook.io/docs/images 

Could anyone provide me an example about how to embed images based on dev-1.0? Thanks.

**Comment by singerdmx** [comment]
Check out https://github.com/memspace/zefyr/issues/469 or https://github.com/singerdmx/flutter-quill/issues/12

**Comment by bluesky2091** [comment]
> Check out #469 or [singerdmx/flutter-quill#12](https://github.com/singerdmx/flutter-quill/issues/12)

Thank you so much for your quick reply! 

What's the major difference between zefyr and flutter-quill? Is flutter-quill built on top of zefyr?  I noticed flutter-quill is not web compatible, is it because of the image picker dependency? Thanks again!

**Comment by singerdmx** [comment]
Note that flutter quill is 100% compatible with quilljs, see below:

INLINE, // refer to https://quilljs.com/docs/formats/#inline
BLOCK, // refer to https://quilljs.com/docs/formats/#block
EMBEDS, // refer to https://quilljs.com/docs/formats/#embeds

It is supposed that react-quill https://github.com/zenoamaro/react-quill is used for web.
Note that the underlying data format are different, meaning you cannot use Zefyr's data in flutter-quill. See https://github.com/singerdmx/flutter-quill/blob/master/app/assets/sample_data.json for the difference.

**Comment by singerdmx** [comment]
@bluesky2091 Are you actually asking about embed images in web?

**Comment by bluesky2091** [comment]
> @bluesky2091 Are you actually asking about embed images in web?

I wrote a flutter app which shares the same code for both mobile app and web. It seems that I can choose zefyr or flutter-quill for mobile version text editor; but so far I haven't figure out a way to have an equivalence for web. 

To your question: Yes, actually I am mainly concerning about web. 

Quick question about react-quill: is that possible to embed it to my flutter app (i can use it only for web)? Thanks!

**Comment by singerdmx** [comment]
`my flutter app (i can use it only for web)` Do you mean flutter web? If so, the answer is No

**Comment by singerdmx** [comment]
I don't understand why so many people start using flutter for web development.
Isn't React the main stream for web development?
I totally understand flutter is the new star replacing react native when it comes to mobile developmemt.
But for web, it is not.

**Comment by bluesky2091** [comment]
> `my flutter app (i can use it only for web)` Do you mean flutter web? If so, the answer is No

yeah, flutter web unfortunately :( I feel that I could just get stuck for the images support in the text editor for flutter web.

**Comment by singerdmx** [comment]
Yeah, I guess maybe you can ask Image Picker https://pub.dev/packages/image_picker to support flutter web.
Then it solves your problem.
You can create an issue there https://github.com/flutter/plugins

**Comment by bluesky2091** [comment]
> https://github.com/flutter/plugins

I am actually looking at this web version image picker: https://pub.dev/packages/image_web_picker but haven't got there to try with zefyr or flutter-quill and see if it works...

**Comment by singerdmx** [comment]
Oh then you can submit a pull request.
It seems that just to make `if else` for using different picker would suffice.

**Comment by bluesky2091** [comment]
Yeah, that sounds great! Thank you so much @singerdmx !

**Comment by singerdmx** [comment]
@bluesky2091  check out https://github.com/singerdmx/flutter-quill/issues/16

**Comment by singerdmx** [comment]
@bluesky2091 
we are adding image support now

@rish07 is enabling web support for [flutter-web](https://github.com/singerdmx/flutter-quill)
It is not yet done fully but I think you can give it a try.
Please use branch master.

**Comment by singerdmx** [comment]
@bluesky2091 web support for image is in. You can try

**Comment by singerdmx** [comment]
Check out https://github.com/singerdmx/flutter-quill/commit/14a417cbf388f9a804b38c5a38bbef933343021f

**Comment by bluesky2091** [comment]
> Check out [singerdmx/flutter-quill@14a417c](https://github.com/singerdmx/flutter-quill/commit/14a417cbf388f9a804b38c5a38bbef933343021f)

That's wonderful! I'll try it today. Thank you so much for this @singerdmx

**Comment by bluesky2091** [comment]
@singerdmx I've switched to flutter-quill and been able to embed images on web. Thanks for your help! 

I noticed an issue that any over-long image (width: 1143 height: 16509) cannot display well there and I believe anyplace inside QuillEidor needs to be adjusted. Could you give me some hint where I need to look into? Thanks! Here is what the image looks like: 

![image](https://user-images.githubusercontent.com/1494005/107888063-759ac280-6ed8-11eb-94e7-be80f4964cce.png)

**Comment by singerdmx** [comment]
@bluesky2091 check out https://bulletjournal1024.slack.com/archives/C0189N1LF24/p1613156828005800 and https://bulletjournal1024.slack.com/archives/C0189N1LF24/p1613305414001200

**Comment by singerdmx** [comment]
@bluesky2091  checkout https://github.com/singerdmx/flutter-quill/pull/24

**Comment by adrian-moisa** [comment]
> singerdmx/flutter-quill#12

I'm one of the people using flutter for web. I've been working with react and angular for the past 8 years. 

For me:

- Reason 1: I'm building a web app that I want to deploy on all possible platforms. I've starting coding from the web platform first. So far I've managed to tick most of the boxes that interest me. I'm building a fully adaptive layout that can handle narrow screens as mobiles (including perf optimisation, not to waste resources). I regularly check the mobile builds if they work as expected.
- Reason 2:  Flutter has superior type safety and refactoring tooling. 
- Reason 3: The tooling to deploy the app on mobile platforms is far more reliable compared to React Native

I'm aware that for the web you don't get the sweet SEO bounty that raw HTML gives you. However for this I plan to render the app with a basic server side rendering app for the unlogged users.
